lib/order.rb

```diff
@@ -0,0 +1,90 @@
+#Products structure { "banana" => 1.99, "cracker" => 3.00 }
+#Fulfuillment_status: :pending, :paid, :processing, :shipped, or :complete
+require_relative 'customer.rb'
+
+class Order
+attr_accessor :products, :customer, :fulfillment_status
+attr_reader :id
+  def initialize(id, products, customer, fulfillment_status= :pending)
+    @id = id
+    @products = products
+    @customer = customer
+
+    #Verifies the fulfillment status is one that matches expected input
+    if [:pending, :paid, :processing, :shipped,:complete].include?(fulfillment_status)
+      @fulfillment_status = fulfillment_status
+    else
+      raise ArgumentError,"A valid fulfillment_status was not put into the Order"
+    end
+  end
+
+  #Adds the total cost of the products and includes tax (7.5%)
+  def total
+    if @products.length == 0
+      return 0
+    else
+      total = @products.values.reduce(:+)
+      total*= 1.075 #Tax
+      total = total.round(2)
+      return total
+    end
+  end
+
+  #Add's a product to the product hash in the order if it doesn't already exist
+  def add_product(product_name, price)
+    if @products.keys.include?(product_name)
+      raise ArgumentError,"The product is already in the order"
+    else
+      @products[product_name] = price
+    end
+  end
+
+  #Removes a product from the product hash in the order if it already exists
+  def remove_product(product_name)
+    if @products.keys.include?(product_name)
+      @products.reject! { |k| k == product_name }
+    else
+      raise ArgumentError, "The product does not exist in the order"
+    end
+  end
+
+  #Method to create the product Hash from CSV input structure
+  def self.product_hash(order)
```

Comment:
This doesn't need to be a class method, as it is only called within the class itself.